astropy/table/tests/test_mixin.py

```diff
@@ -102,14 +104,41 @@ def test_io_ascii_write():
             t.write(out, format=fmt['Format'])
 
 
-def test_io_write_fail(mixin_cols):
+def test_io_Quantity_write(tmpdir):
     """
-    Test that table with mixin column cannot be written by io.votable,
-    io.fits, and io.misc.hdf5
-    every pure Python writer.  No validation of the output is done,
+    Test that table with Quantity mixin column can be written by io.votable,
+    io.fits, and io.misc.hdf5. No validation of the output is done,
     this just confirms no exceptions.
     """
+    t = QTable()
+    t['a'] = Quantity([1,2,4], unit='Angstrom')
+
+    filename = tmpdir.join("table-tmp").strpath
+    open(filename, 'w').close()
+
+    for fmt in ('fits', 'votable', 'hdf5'):
+        if fmt == 'hdf5' and not HAS_H5PY:
+            continue
+        out = StringIO()
+        if fmt == 'hdf5':
+            t.write(filename, format=fmt, path='data', overwrite=True)
+            continue
+        t.write(filename, format=fmt, overwrite=True)
+        if fmt == 'fits':
+            assert QTable.read(filename, format=fmt)['a'].unit == 'Angstrom'
```

Comment:
For this reading test, it is always good to be as explicit as possible, so I would do
```
qt = QTable.read(...)
assert isinstance(qt['a'], Quantity)
assert qt['a'].unit == 'Angstrom'
```